### ISSUE TITLE
Unit tests don't run on master via ./run_tests and docker

### ISSUE THREAD

**Issue by kevinclark** [dense-analysis/ale#3448]
On 681a6e371d02 (which is top of master at the moment). Logs [here](https://gist.github.com/kevinclark/bc457f247940dd80c87c93149ddadde4). Though I ran it with `--fast -q` in the log, it fails without those flags as well.

I'm on macOS 10.15.7. Docker version 19.03.13, build 4484c46d9d. Let me know what other versions might matter. I see CI is passing and figured docker should be isolating the environment, but 🤷‍♂️

**Comment by kevinclark** [comment]
Looks like there's a few issues. One is that many of the test scripts run grep/sed/ls/sort using the local system instead of the docker instance and then you get slight differences in implementation that causes failures. Another is that the tag reference check is using a unified diff to look for invalid references, but it just does it based on a '+' line, which doesn't allow for reordering. F.e:

```diff

-ALEComplete
 ALECompletePost
-ALECompletePost-autocmd
+ALEComplete
```

Here's a case where `ALEComplete` shows up at an invalid reference when it's just out of expected order.

I'm working up a PR to resolve these issues since it blocks my work on #3470. It's going to result in some of the doc-sorting enforcement changing order slightly, but it'll be consistent across platforms.

**Comment by hsanson** [comment]
@kevinclark can this be closed now that #3471 is merged?

**Comment by kevinclark** [comment]
Yep!
